src/condvar.rs

```diff
@@ -350,10 +351,10 @@ impl Condvar {
         }
 
         // ... and re-lock it once we are done sleeping
-        if result == ParkResult::Unparked(TOKEN_HANDOFF) {
-            unsafe { deadlock::acquire_resource(mutex as *const _ as usize) };
-        } else {
-            mutex.lock();
+        match result {
+            ParkResult::Unparked(TOKEN_HANDOFF) => unreachable!("can't be handed off"),
```

Comment:
`TOKEN_HANDOFF` is actually reachable if we are requeued onto a mutex and then another unlocks that mutex with `unlock_fair`.